apps/status-api/src/controllers/OverallStatusController.ts

```diff
@@ -0,0 +1,40 @@
+import { Controller, Get } from '@nestjs/common'
+import { WhaleApiClient } from '@defichain/whale-api-client'
+import { BlockchainStatus, BlockchainStatusController } from './BlockchainStatusController'
+import { WhaleApiProbeIndicator } from '../modules/WhaleApiModule'
+
+@Controller('overall')
+export class OverallStatusController {
+  constructor (
+    private readonly client: WhaleApiClient,
+    private readonly probe: WhaleApiProbeIndicator,
+    private readonly blockchainStatusController: BlockchainStatusController
+  ) {
+  }
+
+  /**
+   *  To provide overall status for Ocean and blockchain services. Returns
+   *  'operational' when both services are up,
+   *  'outage' when either service is down and
+   *  'degraded' when blockchain service is degraded.
+   *
+   *  @return {Promise<{ status: BlockchainStatus }>}
+   */
+
+  @Get()
+  async getAggregateStatus (): Promise<{ status: BlockchainStatus }> {
+    const liveness = await this.probe.liveness()
+
+    if (liveness.whale.status === 'down') {
+      return {
+        status: 'outage'
+      }
+    }
```

Comment:
On top of the liveness, can we also query readiness to check for the status?

Reply:
Done!